### ISSUE TITLE
Update C++ code to conform to our coding standards

### ISSUE THREAD

**Issue by prock-fife** [fifengine/fifengine#550]
See [http://wiki.fifengine.net/Coding_standards] for more information.

**Comment by jakoch** [comment]
- [ ] create `.astylerc` or `.clang-format` to enfore the coding-style and formatting via a tool

**Comment by jakoch** [comment]
I've configured an `.astylerc` file. But i'm not sure, if this formatting is wanted. 
The bracket style is different and tabs are converted to spaces for C++ code.
Let's discuss this a bit and tweak it as necessary.

You can use a command like the following to test the formatter on the C++ stuff of `engine\core`:
`C:\astyle\bin\AStyle.exe d:\2-github.com\fifengine\fifengine\engine\core\*.* -n --options=.astylerc`

`.astylerc`:

```
#--------------------------------------------------
# AStyle Configuration
#--------------------------------------------------

# use K&R formatting
style=kr

indent=spaces=4

#--------------------------------------------------
# Bracket Modify Options
#--------------------------------------------------

# add bracket to unbracketed one line conditional statement(e.g. 'if','for','while').
add-brackets

attach-namespaces

attach-classes

attach-inlines

attach-extern-c

#--------------------------------------------------
# Indentation Options
#--------------------------------------------------

indent-namespaces

indent-classes

indent-modifiers

indent-switches

indent-cases

indent-preprocessor

indent-preproc-block

indent-preproc-cond

indent-col1-comments

min-conditional-indent=2

max-instatement-indent=80

#--------------------------------------------------
# Padding Options
#--------------------------------------------------

pad-oper

pad-header

unpad-paren

#--------------------------------------------------
# Formatting options
#--------------------------------------------------

break-blocks

align-pointer=type

delete-empty-lines

# convert tabs into spaces
convert-tabs

# break line, if the code exceed 100 columns.
max-code-length=100

break-after-logical

mode=c

#--------------------------------------------------
# Processing
#--------------------------------------------------

# travel through dirs
recursive

# show only formatted files
formatted
```

**Comment by jakoch** [comment]
For the formatting of Python code, we could adopt and enforce the style used by UH:
https://github.com/unknown-horizons/unknown-horizons/blob/master/.pylintrc

**Comment by prock-fife** [comment]
> For the formatting of Python code, we could adopt and enforce the style used by UH

Agreed.  For the C++ stuff our standards are listed here: https://github.com/fifengine/fifengine/wiki/coding-standards

I'm open to update the standards if the need requires.

**Comment by jakoch** [comment]
My `.astylerc` clashes a bit with the requirements of the already existing coding standard for C++.

**tabs vs. spaces**
I think the biggest change i introduced is the switch from tabs to spaces.
I don't want to get into a big debate on pro's and con's of tabs vs. spaces here, 
because my goal is to format the code consistently with either tabs or spaces,
but not a mix of them, which looks terrible https://github.com/fifengine/fifengine/blob/ad585a21430136be96b332da712f5d3f31083f0a/engine/core/modules.h#L92

This change is basically following Python's PEP-8, where spaces are the preferred indentation method:
https://www.python.org/dev/peps/pep-0008/#tabs-or-spaces

That means the C++ (and the Python code) will use spaces for indentation consistently.
Is the switch to spaces on the project(s) ok for the team?

**Comment by vtchill** [comment]
I prefer spaces over tabs as well. I'm all for the change to spaces.

**Comment by prock-fife** [comment]
Personally it makes no difference.  I'm sure there are arguments either way.   We do use spaces in the python code so it would be nice to be consistent.  Are there any caveats to making the change?  It will require a modification of every line of code.

**Comment by helios2000** [comment]
The goal to have a consistently code is great. My problem with the tab to space change is that we lost the history of the code. So github functions like blame would bring nothing more.
We currently use tabs in C++ and Python code. So I would prefer to change the few remaining spaces to tabs.

**Comment by Beliaar** [comment]
Just a note, I just found this:

https://coderwall.com/p/x8xbnq/git-don-t-blame-people-for-changing-whitespaces-or-moving-code.
